src/blm.jl

```diff
@@ -0,0 +1,87 @@
+abstract type InferenceStrategy end
+struct MCMC{TAlg,TArgs,TKwArgs} <: InferenceStrategy
+    nsamples::Int
+    alg::TAlg
+    algargs::TArgs
+    algkwargs::TKwArgs
+    MCMC(alg::TAlg, nsamples::Int, args...;kwargs...) where {TAlg<:Turing.InferenceAlgorithm} =
+        new{TAlg,typeof(args),typeof(kwargs)}(nsamples,alg,args,kwargs)
+end
+# TODO add VI support
+# struct VI <: InferenceStrategy end
+
+export InferenceStrategy, MCMC
+
+# Default interface methods
+"""
+    inference(::DynamicPPL.Model, strat::InferenceStrategy)
+
+Run probabilistic inference for the given model using the given InferenceStrategy (e.g. MCMC or variational inference).
+The type of the result depends on the implementation. MCMC implementations should return a ChainDataFrame, whereas VI
+should return a fitted variational distribution.
+"""
+inference(::DynamicPPL.Model, strat::InferenceStrategy) = error("no inference implementation for $(typeof(strat))")
+"""
+    preprocess(::InferenceStrategy, f::FormulaTerm, df::DataFrame)
+
+Applies any necessary preprocessing to the user-provided formula and DataFrame. Can be overrided for specific
+inference strategies if necessary. Should return a data tuple (y,X) representing the response and design matrix respectively.
+"""
+preprocess(::InferenceStrategy, f::FormulaTerm, df::DataFrame) = modelcols(f, df)
+"""
+    postprocess(::InferenceStrategy, inferenceresult, f::FormulaTerm, df::DataFrame)
+
+Applies postprocessing to the given inference result. Returns the (possibly modified) inference result.
+"""
+postprocess(::InferenceStrategy, inferenceresult, f::FormulaTerm, df::DataFrame) = inferenceresult
+
+# MCMC inference
+inference(model::DynamicPPL.Model, mcmc::MCMC) = sample(model, mcmc.alg, mcmc.nsamples, mcmc.algargs...;mcmc.algkwargs...)
+
+function postprocess(::MCMC, chain::MCMCChains.Chains, f::FormulaTerm, df::DataFrame)
+    # Get formula names.
+    nms = coefnames(f.rhs)
+    nms_dict = Dict(["β[$i]" => "β[$(nms[i])]" for i in eachindex(nms)])
+    # Overwrite the default names.
+    chain = MCMCChains.replacenames(chain, nms_dict)
+end
+
+"""
+    blm(f::FormulaTerm, df::DataFrame, inferencestrat::InferenceStrategy=MCMC(NUTS(),1000); alphaprior, betaprior, sigmaprior)
+
+Fits a Bayesian linear model specified by `f` to the given data. Return value depends on the specified inference strategy.
+MCMC will return a ChainDataFrame, whereas VI will return a fitted variational distribution.
+"""
+function blm(f::FormulaTerm, df::DataFrame, inferencestrat::InferenceStrategy=MCMC(NUTS(),1000);
+    alphaprior=nothing, betaprior=nothing, sigmaprior=nothing)
```

Comment:
```suggestion
function blm(
    f::FormulaTerm,
    df::DataFrame, 
    inferencestrat::InferenceStrategy=MCMC(NUTS(), 1000);
    alphaprior=nothing, 
    betaprior=nothing, 
    sigmaprior=nothing
)
```
Mild ease of reading suggestion.

Reply:
Has julia coalesed around something like `black` (e.g. [JuliaFormatter](https://github.com/domluna/JuliaFormatter.jl))?

Reply:
I think in general we tilt towards [BlueStyle](https://github.com/invenia/BlueStyle).

Reply:
Cool, I actually really prefer this, but I got the impression that the Julia community shied away from doing this kind of multi-line indenting? I'm glad to be wrong, though!